src/screen/title.rs

```diff
@@ -0,0 +1,47 @@
+use bevy::prelude::*;
+
+use super::Screen;
+use crate::util::ui::*;
+
+pub(super) fn plugin(app: &mut App) {
+    // Screen setup and teardown.
+    app.add_systems(OnEnter(Screen::Title), enter_title)
+        .add_systems(OnExit(Screen::Title), exit_title);
+
+    app.add_systems(Update, handle_title_action.run_if(in_state(Screen::Title)));
+}
+
+#[derive(Component, PartialEq, Eq)]
+enum TitleAction {
+    Play,
+    Credits,
+}
+
+fn enter_title(mut commands: Commands) {
+    let list = commands.my_root().insert(StateScoped(Screen::Title)).id();
+
+    commands
+        .my_button("Play")
+        .insert(TitleAction::Play)
+        .set_parent(list);
+    commands
+        .my_button("Credits")
+        .insert(TitleAction::Credits)
+        .set_parent(list);
+}
+
+fn exit_title() {}
```

Comment:
🙃

Reply:
This is just for consistency but I can definitely remove it. State-scoped entities making exit systems redundant :weary:

Reply:
Done.

Reply:
It's good for clearing resources which cannot be scoped yet

Reply:
but yeah, adding that empty function is pedantic 😄